### PR TITLE
Runs page empty states

### LINKED ISSUE #681: A project shouldn't show an empty Runs table
When I create a new project, I shouldn't see an empty Runs table. It should be hidden  until files have been added to the project.

screen shot 2015-06-02 at 5 55 20 pm

### PR DESCRIPTION
**Pull request by jaclynperrone**
This commit does a couple things:

1. Adds empty states for 0 Bams and 0 Runs on the project table (closes #681)
2. Adds empty state for 0 recent comments
3. Adds the updated pdiff screenshots

screen shot 2015-09-18 at 4 52 20 pm

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/cycledash/844)

### CODE REVIEW

File: tests/ENV.sh [hunk 1]
```diff
@@ -9,3 +9,4 @@ export CELERY_BACKEND='db+sqlite:///celery-dpxdt.db'
 export CELERY_BROKER='amqp://localhost'
 export SECRET_KEY='TESTSECRETS'
 export BCRYPT_LOG_ROUNDS=1
+export TYPEKIT_URL="//use.typekit.net/vml3kel.js"
```
Comment (ihodes) [inline tests/ENV.sh]: This shouldn't be in the public repo (this is my private key)

File: cycledash/static/js/runs/components/RunsPage.js [hunk 2]
```diff
@@ -179,14 +179,16 @@ var ProjectTable = React.createClass({
...
       table = <BamsTable bams={this.props.bams}
                          selectedBamId={this.state.selectedBamId}
                          createClickBamHandler={this.createClickBamHandler} />;
-    } else {
+    } else if (!this.state.bamsTable && numRuns > 0) {
       table = <RunsTable runs={this.props.runs}
                          selectedRunId={this.state.selectedRunId}
                          createClickRunHandler={this.createClickRunHandler} />;
+    } else {
+      table = <div className="project-empty-state">No files uploaded</div>;
```
Comment (ihodes) [inline cycledash/static/js/runs/components/RunsPage.js]: I think it would be better to say either "No runs yet" or "No BAMs yet", depending on the table you're looking at. We're not really expecting runs or BAMs to be "uploaded files".

**Comment by ihodes** [comment]
LGTM % 2 comments!

File: tests/ENV.sh (hunk 1 above)
Comment (jaclynperrone) [inline tests/ENV.sh]: Yikes! My bad. Didn't mean to commit this one. I'll remove it.

**Comment by jaclynperrone** [comment]
@ihodes Back to you! I made the empty states more specific and added a new test to show the empty BAMs table.

screen shot 2015-09-22 at 12 56 16 pm
screen shot 2015-09-22 at 12 56 08 pm

File: cycledash/static/js/comments/components/comment.js [hunk 3]
```diff
@@ -23,11 +23,17 @@ var LatestComments = React.createClass({
     comments: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
   },
   render: function() {
+    var numComments = this.props.comments.length;
+    var recentComments;
+    if (numComments > 0) {
+      recentComments =  <div><h4>Last {numComments} Comments<a href='/comments' className='all-comments'>See all</a></h4>
```
Comment (ihodes) [inline cycledash/static/js/comments/components/comment.js]: Nits: there should be one space after the =, and the JSX should be formatted like HTML (e.g. nesting and new lines).

File: cycledash/static/js/runs/components/RunsPage.js [hunk 4]
```diff
@@ -179,14 +179,18 @@ var ProjectTable = React.createClass({
...
+    if (this.state.bamsTable && numBams > 0) {
       table = <BamsTable bams={this.props.bams}
                          selectedBamId={this.state.selectedBamId}
                          createClickBamHandler={this.createClickBamHandler} />;
-    } else {
+    } else if (!this.state.bamsTable && numRuns > 0) {
       table = <RunsTable runs={this.props.runs}
                          selectedRunId={this.state.selectedRunId}
                          createClickRunHandler={this.createClickRunHandler} />;
+    } else if (this.state.bamsTable && numBams === 0) {
```
Comment (ihodes) [inline cycledash/static/js/runs/components/RunsPage.js]: These two cases (when numBams or rows is 0) can be handled in the RunsTable and BamsTable components instead; that way you avoid a big if/else block here.

**Comment by ihodes** [comment]
Nice! Just two more minor comments.

**Comment by ihodes** [comment]
LGTM! Thanks for the feature.

**Comment by jaclynperrone** [comment]
Woo! No prob!
